tests/Issues/IssueRefreshScope/Fixture/foreach_iterable.php.inc

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Rector\Core\Tests\Issues\IssueRefreshScope\Fixture;
+
+final class ForeachIterable
+{
+    public function run(): void
+    {
+        $itemsByName = $this->getItemsByName();
+        foreach ($itemsByName as $values) {
+            if (count($values)) {
+                continue;
+            }
+        }
+    }
+
+    /**
+     * @return array<string, string[]>
+     */
+    private function getItemsByName(): array
+    {
+        return [
+            'name' => ['Tom', 'Matthias'],
+        ];
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Core\Tests\Issues\IssueRefreshScope\Fixture;
+
+final class ForeachIterable
+{
+    public function run(): void
+    {
+        $itemsByName = $this->getItemsByName();
+        foreach ($itemsByName as $items) {
+            if (count($items)) {
+                continue;
+            }
+        }
```

Comment:
Before this change, the type was lost on `$values` renamed to `$items`.
Now the `$items` have the `ArrayType` as original `$values` have :+1: